flextape/service/queue.go

```diff
@@ -0,0 +1,137 @@
+package service
+
+import (
+	"sort"
+)
+
+// invocationq is a simple queue of invocations.
+//
+// Beside implementing the classic Enqueue and Dequeue methods:
+// - It partially implements the sort.Interface to allow the easy
+//   implementation of invocation prioritization policies.
+// - It maintains per-entry state (QueueID) so it's easy to tell
+//   the position of each invocation by just looking at that value.
+type invocationq []*invocation
```

Comment:
nit: This name looks like a typo of `invocation` - can it be `invocationQueue` or at least `invocationQ`, which are at least visibly distinct from `invocation`?

Reply:
done